plugins/scaffolder-backend/src/lib/repo/index.ts

```diff
@@ -0,0 +1,50 @@
+import { Repository as DiskRepository } from './disk';
+
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export interface Template {
+  id: string;
+  name: string;
+  description: string;
+  ownerId: string;
+}
+
+export abstract class RepositoryBase {
+  // lists all templates available
+  abstract async list(): Promise<Template[]>;
+  // can be used to build an index of the available templates;
+  abstract async reindex(): Promise<void>;
+  // returns a directory to run the templaterin
+  abstract async prepare(id: string): Promise<string>;
+}
+
+class RepositoryImplementation implements RepositoryBase {
+  repo?: RepositoryBase;
+
+  constructor() {
+    this.repo = DiskRepository;
+  }
+
+  public setRepository(repo: RepositoryBase) {
+    this.repo = repo;
+  }
+
+  list = () => this.repo!.list();
+  prepare = (id: string) => this.repo!.prepare(id);
+  reindex = () => this.repo!.reindex();
+}
+
+export const Repository = new RepositoryImplementation();
```

Comment:
Do we have to have this? xP

Would rather provide it through config and default to disk if it's not set. Ties into the `createScaffolder()` comment above.

Reply:
Yeah I get what you mean, `createScaffolder` with the config makes more sense.